### PR TITLE
Added clear button and refractored code

### PR DESCRIPTION
**Pull request by ravjot07**
Added clear button that Clears the canvas and reset the drawing lines. Contribution towards #14 

https://github.com/user-attachments/assets/d7fd9ea0-9934-4dca-8816-cda3de80121b

### CODE REVIEW

**Comment by ravjot07** [comment]
@quozl can you review this pr

File: js/activity.js [hunk 1]
```diff
@@ -2,32 +2,43 @@ define(function (require) {
...
         });
+
+        // Handle New Button Click
         var newButton = document.getElementById("new-button");
         newButton.onclick = function () {
             new_positions();
         }
-        // Make the activity stop with the stop button.
+
+        // Handle Stop Button Click
```
Comment (chimosky) [inline js/activity.js:25]: Comments like this are unnecessary, and not needed.
Please remove them.

File: css/activity.css [hunk 2]
```diff
@@ -5,3 +5,7 @@
 #main-toolbar #new-button {
   background-image: url(../icons/new-button.svg);
 }
+
+#main-toolbar #clear-button {
+  background-image: url(../icons/clear-button.svg);
+}
```
Comment (chimosky) [inline css/activity.css:11]: You can remove the No new line at EOF by running `echo >> css/activity.css`, this makes it better to view when using diff tools or cat.

File: js/activity-works.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 define(function (require) {
     var activity = require("sugar-web/activity/activity");
     var icon = require("sugar-web/graphics/icon");
+    require("easel");
+    require("handlebars");
+    var shapes = require("activity/shapes");
```
Comment (chimosky) [inline js/activity-works.js:6]: Are these used here?

File: js/activity-works.js [hunk 4]
```diff
@@ -41,231 +54,247 @@ define(function (require) {
...
-                // enable touch interactions if supported on the current device:
-                createjs.Touch.enable(stage);
-
-                // keep tracking the mouse even when it leaves the canvas
-                stage.mouseMoveOutside = true;
-
-                // enabled mouse over / out events
-                stage.enableMouseOver(10);
+        // Get things started
+        init();
```
Comment (chimosky) [inline js/activity-works.js:68]: Please remove redundant comments like this, they're not useful.

File: js/activity.js [hunk 5]
```diff
@@ -2,32 +2,43 @@ define(function (require) {
...
         });
+
+        // Handle Clear Button Click
+        var clearButton = document.getElementById("clear-button"); // **New Line**
+        clearButton.addEventListener('click', clearCanvas); // **New Line**
+
         // Then create a list of the label elements
-        nlabels = [];
-        for (i = 0; i < 21; i++) {
+        var nlabels = [];
```
Comment (chimosky) [inline js/activity.js:36]: If this is consistent throughout this code, there's no need to change it.

**Comment by chimosky** [comment]
See #15 to make sure you've addressed the concerns listed there, reviewed, not tested.

The code refactoring shouldn't be in the same commit as the clear button change, keep both separate as it's easier to review that way.

File: js/activity.js (hunk 1 above)
Comment (ravjot07) [reply js/activity.js:25]: i have removed unnecessary comments

File: css/activity.css (hunk 2 above)
Comment (ravjot07) [reply css/activity.css:11]: done...

File: js/activity-works.js (hunk 4 above)
Comment (ravjot07) [reply js/activity-works.js:68]: done..

File: js/activity.js (hunk 5 above)
Comment (ravjot07) [reply js/activity.js:36]: sure i will keep them as it is....

File: js/activity.js [hunk 6]
```diff
@@ -2,32 +2,43 @@ define(function (require) {
...
+        for (var i = 0; i < 21; i++) {
```
[the remark this answers is not in the document]
Comment (ravjot07) [reply js/activity.js:37]: done...

**Comment by ravjot07** [comment]
> See #15 to make sure you've addressed the concerns listed there, reviewed, not tested.
> 
> The code refactoring shouldn't be in the same commit as the clear button change, keep both separate as it's easier to review that way.

yes  i have addressed the concerns listed in #15.... mainly they discussed about position of the button and its icon visibility and its functioning  about deleting only the drawing rather than whole canvas...... i have covered all of that

**Comment by ravjot07** [comment]
@chimosky are there any more changes or suggestion or we should merge this PR

**Comment by chimosky** [comment]
Would be great to have better commit messages than "Minor Updates", see [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).

**Comment by ravjot07** [comment]
> Would be great to have better commit messages than "Minor Updates", see [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).
> 
> I'll test this once I can.

@chimosky sure i will take care of commit msg next time.

**Comment by quozl** [comment]
You can and should change commit messages now rather than wait for next time.

**Comment by ravjot07** [comment]
> You can and should change commit messages now rather than wait for next time.

Done @quozl

**Comment by quozl** [comment]
Tested.  Reviewed.

**Pull request merged**